WordPress/WordPressTest/AccountSettingsRemoteInterfaceStub.swift

```diff
@@ -0,0 +1,77 @@
+@testable import WordPress
+import WordPressKit
+
+class AccountSettingsRemoteInterfaceStub: AccountSettingsRemoteInterface {
```

Comment:
Not a blocker to this PR, but my personal preference has been not implementing a stub like this, mainly because there are lots of boilerplate code to write and maintain. It seems easier to me to use `OHHTTPStubs` to stub HTTP requests.

It's a totally a different story if Swift provides a way to stub protocol methods like `AccountSettingsRemoteInterface.stub().function(getSettings(_:)).returns(AccountSettings())`. But unfortunately that's not reality yet.

However, I wonder if we can use the new macro feature to eliminate the boilerplate. That'd be a fun thing to explore.

Reply:
Ah that's interesting. My preference is the opposite. That is, I really like having stubs like this (which maybe is not surprising given how I wrote the PR 😅 )

It's totally true that we have additional coding to do, but I see it as a one-time cost (or one-per-interface-change, maybe) which is spread across hundreds of test runs. What we get from those kind of test doubles is a more unit level type of testing, I think, because they truly isolate the SUT.

Of course, I'm not trying to say that they are _the best_ approach. There's no such thing. Alongside with more coding, using dedicated new objects as test doubles also means that the configuration tested at test-time is not the same as run-time. One could argue the previous solution with OHHTPStubs uses a stack that is closer to the "real" runtime usage. I could keep arguing the pros-and-cons for a while, but I guess is best if I stop here. 😄 

---

And 💯 on the hope that Swift Macros will make this approach a bit easier.